syntax = "proto3";

package proto;

enum ServerErrorReason {
  SERVER_ERROR_REASON_UNKNOWN = 0;
  SERVER_UNEXPECTED_EXCEPTION = 1;
  SERVER_INTERRUPTED = 2;
  SERVER_PERSISTENCE_ERROR = 3;
  SERVER_ERROR_SET_IN_RESPONSE_BY_ACTION = 4;
  SERVER_ERROR_NOT_SEEDED_GUICE_EXCEPTION = 5;
  SERVER_NOT_A_TRANSACTION_ACTION_EXCEPTION = 6;
  SERVER_MESSAGE_INSTANTIATION_EXCEPTION = 7;
  SERVER_NOT_IMPLEMENTED_EXCEPTION = 8;
  SERVER_EXPERIMENT_DIVERSION_FAILURE = 9;
  SERVER_EXPERIMENT_OVERRIDE_FAILURE = 10;
  SERVER_REMOTE_FRAMEWORK_ERROR = 11;
  SERVER_TIMEOUT_ERROR = 12;
  SERVER_RESPONSE_CAN_NOT_BE_THE_DEFAULT_INSTANCE = 13;
  SERVER_INVALID_AUTH_CONTEXT = 14;
}
